design/baremetal-operator/configmap_for_host_inspection_data.md

```diff
@@ -0,0 +1,186 @@
+    cluster.
+
+2. `clusterctl.cluster.x-k8s.io=""` and `cluster.x-k8s.io/provider="metal3"`
+    labels on InspectionData CRD to extend the list of provider objects.
+
+3. `clusterctl.cluster.x-k8s.io/move` label on InspectionData CRD. This will
+    not move child objects as in option 1.
+
+### Implementation Details/Notes/Constraints
+
+The life of an InspectionData is tightly coupled to a BareMetalHost. Creating or
+deleting a BareMetalHost will result in creation or deletion of a InspectionData
+respectively. Deprovisioning of the BareMetalHost will result in the inspection
+data removal from the spec of InspectionData.
+
+**Note:** [cross-namespace owner references are not allowed](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications). Thus, InspectionData
+and BareMetalHost must be in the same namespace.
+
+- BareMetalHost is created
+- Operator creates a InspectionData with the same name and namespace as
+    BareMetalHost and sets ownerReferences
+- Provisioning is requested, operator fills out the InspectionData with the
+    host inspection data
+- Pivoted to a target cluster, BMO restores host inspection data from InspectionData
+- Deprovisioning is requested, operator cleans up the inspection data from the
+    InspectionData
+- BareMetalHost deletion requested, operator deletes the InspectionData and
+    BareMetalHost
+
+Operator already respects `baremetalhost.metal3.io/paused` annotation if set on
+a BareMetalHost and stops the reconciliation for the BareMetalHost. This was
+added as part of Cluster API (CAPI) workflow prerequisite, so that if a CAPI Cluster
+object is paused, the provider is capable of pausing all the child objects like
+BareMetalHost. Since InspectionData is dependent on BareMetalHost, we need to
+ensure that when Cluster is paused, we pause reconciliation of InspectionData too.
+
+The spec of InspectionData:
+
+```go
+// Adding HardwareDetails struct here for visibility. During implementation,
+// we don't have to duplicate it, but use from baremetalhost_types.go.
+type HardwareDetails struct {
+
+    SystemVendor HardwareSystemVendor `json:"systemVendor,omitempty"`
+    Firmware     Firmware             `json:"firmware,omitempty"`
+    RAMMebibytes int                  `json:"ramMebibytes,omitempty"`
+    NIC          []NIC                `json:"nics,omitempty"`
+    Storage      []Storage            `json:"storage,omitempty"`
+    CPU          CPU                  `json:"cpu,omitempty"`
+    Hostname     string               `json:"hostname,omitempty"`
+
+}
+
+type InspectionDataSpec struct {
+    // The hardware discovered on the host.
+    HardwareDetails *HardwareDetails `json:"hardware,omitempty"`
+}
+```
+
+The status of InspectionData:
+
+```go
+type InspectionDataStatus struct {
+    // Last time the status was updated/observed.
+    // +optional
+    LastUpdated *metav1.Time `json:"lastUpdated,omitempty"`
```

Comment:
I'm +1 on immutability point and based on that it makes sense to drop `LastUpdated`, because there is no possibility anyways updating the CR due to immutability.

Reply:
ow you meant with RBAC, I was rather thinking to do it with ValidationgWebhook.

Reply:
We can do both :)